library/block_cipher.c

```diff
@@ -42,6 +83,17 @@ void mbedtls_block_cipher_free(mbedtls_block_cipher_context_t *ctx)
 int mbedtls_block_cipher_setup(mbedtls_block_cipher_context_t *ctx,
                                mbedtls_cipher_id_t cipher_id)
 {
+#if defined(MBEDTLS_BLOCK_CIPHER_SOME_PSA)
+    if (psa_can_do_cipher(cipher_id)) {
```

Comment:
That's due to the way I implemented it: since I was experimentally developing the feature, I first wrote all the code and then split it into commits. As a consequence it might happen that the code cannot be built in some intermediate commit. This works fine because in the end we're accepting the PR entirely or not and not just picking some commits from it. I understand that this might not be the best way to create git history for a reviewer though

Reply:
For review it was not a problem except for the initial surprise. Way easier than reviewing a single massive commit anyway :)

However, in general we try to structure commits so that each one at least builds, in order to facilitate bisecting should we need it in the future. But that's not a hard rule and I don't think it's worth the effort restructuring here. (Also, git bisect has provisions for skipping commits that can't be tested, it just takes a bit more care.)